src/SFA.DAS.EmployerIncentives.Domain/ApprenticeshipIncentives/ApprenticeshipIncentive.cs

```diff
@@ -256,6 +261,32 @@ public void ValidateDaysInLearning(Guid pendingPaymentId, Learner matchedLearner
             pendingPayment.AddValidationResult(PendingPaymentValidationResult.New(Guid.NewGuid(), collectionPeriod, ValidationStep.HasDaysInLearning, hasEnoughDaysInLearning));
         }
 
+        public void PauseSubsequentPayments(ServiceRequest serviceRequest)
+        {
+            if (Model.PausePayments == false)
+            {
+                Model.PausePayments = true;
+                AddEvent(new PaymentsPaused(Model.Account.Id, Model.Account.AccountLegalEntityId, Model, serviceRequest));
+            }
+            else
+            {
+                throw new PausePaymentsException("Payments already paused");
```

Comment:
Do we care?  To make it idempotent just ignore since the current state is correct.

Reply:
No, we do care. It's not idempotent as a user is entering information and applying it to this incentive payment. It's quite likely they are updating these payments on mass and if they might have already updated this apprenticeship. The user will be under the impression that their update was successful, but the audit information could be wrong.

Reply:
But the update would be successful.  Isn't it the outcome that is important not the process?

Reply:
i agree with @Stephek, those checks are unnecessary and idempotence should be encouraged. Unless there is a specific requirement for this use case to produce an error ❓

also, afaik, EF will take care of the "dirty-tracking" so there won't be a redundant call to the database made 🟢

🎄